weighted_graph.py

```diff
@@ -55,35 +56,75 @@ def del_node(self, node):
             # Delete the node from any neighbor lists it is in.
             for value in self.graph.itervalues():
                     try:
-                        value.remove(node)
-                    except ValueError:
+                        del value[node]
+                    except KeyError:
                         pass
         except KeyError:
             raise KeyError('{} not in graph'.format(node))
 
     def has_node(self, node):
-        ''' Return True or False ifthe node is in the graph or not. '''
+        ''' Return True or False if the node is in the graph or not. '''
         return node in self.graph
 
     def neighbors(self, node):
         ''' Return the neighbors of a node. '''
         try:
-            return self.graph[node]
+            return self.graph[node].keys()
         except KeyError:
             raise KeyError('{} not in graph'.format(node))
 
     def adjacent(self, node1, node2):
         ''' Check is if two nodes have an edge connecting them. '''
         try:
-            return node2 in self.graph[node1]
+            return node2 in self.graph[node1].keys()
```

Comment:
You don't have to render a list of keys, membership in a dictionary will work here.  `node2 in self.graph[node1]`